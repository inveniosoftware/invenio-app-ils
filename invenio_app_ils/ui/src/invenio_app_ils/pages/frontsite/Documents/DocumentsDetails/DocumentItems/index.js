import { connect } from 'react-redux';
import DocumentItemsComponent from './DocumentItems';

const mapStateToProps = state => ({
  documentDetails: state.documentDetailsFront.data,
});


export const DocumentItems = connect(
  mapStateToProps,
  null
)(DocumentItemsComponent);
