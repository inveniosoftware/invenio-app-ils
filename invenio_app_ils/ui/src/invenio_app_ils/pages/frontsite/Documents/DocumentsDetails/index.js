import { connect } from 'react-redux';
import { fetchDocumentsDetails } from './state/actions';
import DocumentsDetailsContainerComponent from './DocumentsDetailsContainer';

const mapDispatchToProps = dispatch => ({
  fetchDocumentsDetails: documentPid =>
    dispatch(fetchDocumentsDetails(documentPid)),
});

const mapStateToProps = state => ({
  isLoading: state.documentsDetails.isLoading,
  data: state.documentsDetails.data,
  hasError: state.documentsDetails.hasError,
});

export const DocumentsDetailsContainer = connect(
  mapStateToProps,
  mapDispatchToProps
)(DocumentsDetailsContainerComponent);
