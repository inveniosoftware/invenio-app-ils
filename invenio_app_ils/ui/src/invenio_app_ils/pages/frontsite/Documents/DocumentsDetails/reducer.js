export { default as documentDetailsFrontReducer } from './state/reducer';
export {
  default as loanRequestFormReducer,
} from './components/LoanRequestForm/state/reducer';
