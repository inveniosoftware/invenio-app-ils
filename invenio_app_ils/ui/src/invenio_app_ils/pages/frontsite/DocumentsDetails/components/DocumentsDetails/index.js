import { compose } from 'redux';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import DocumentsDetailsComponent from './DocumentsDetails';

const mapStateToProps = state => ({
  isLoading: state.documentsDetails.isLoading,
  data: state.documentsDetails.data,
  hasError: state.documentsDetails.hasError,
});

export const DocumentsDetails = compose(
  withRouter,
  connect(mapStateToProps)
)(DocumentsDetailsComponent);
