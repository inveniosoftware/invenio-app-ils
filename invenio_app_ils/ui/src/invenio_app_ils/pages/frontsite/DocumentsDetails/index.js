import { compose } from 'redux';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import { fetchDocumentsDetails } from './state/actions';
import DocumentsDetailsContainerComponent from './DocumentsDetailsContainer';

const mapDispatchToProps = dispatch => ({
  fetchDocumentsDetails: documentPid =>
    dispatch(fetchDocumentsDetails(documentPid)),
});

export const DocumentsDetailsContainer = compose(
  withRouter,
  connect(
    null,
    mapDispatchToProps
  )
)(DocumentsDetailsContainerComponent);
