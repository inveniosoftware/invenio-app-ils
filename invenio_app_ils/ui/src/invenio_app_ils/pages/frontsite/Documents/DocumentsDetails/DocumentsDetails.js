import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Container } from 'semantic-ui-react';
import { DocumentMetadata } from './components';
import { goTo } from '../../../../history';
import { FrontSiteRoutes } from '../../../../routes/urls';
import { SearchBar } from '../../../../common/components/SearchBar';
import { DocumentPanel } from './components/DocumentPanel';
import { Error } from '../../../../common/components/Error';
import { Breadcrumbs } from '../../components/Breadcrumbs';
import { DocumentTags } from './components/DocumentMetadata/components';
import { ILSParagraphPlaceholder } from '../../../../common/ILSPlaceholder';

export default class DocumentsDetails extends Component {
  constructor(props) {
    super(props);
    this.fetchDocumentsDetails = this.props.fetchDocumentsDetails;
    this.state = { searchQuery: '' };
    this.renderElement = props.renderElement || this._renderElement;
  }

  componentDidMount() {
    this.unlisten = this.props.history.listen(location => {
      if (location.state && location.state.documentPid) {
        this.fetchDocumentsDetails(location.state.documentPid);
      }
    });
    this.fetchDocumentsDetails(this.props.match.params.documentPid);
  }

  componentWillUnmount() {
    this.unlisten();
  }

  onSearchClick = () => {
    const query = encodeURIComponent(this.state.searchQuery);
    goTo(FrontSiteRoutes.documentsListWithQuery(query));
  };

  onSearchInputChange = (value, event) => {
    this.setState({ searchQuery: event.target.value });
  };

  breadcrumbs = () => [
    { to: FrontSiteRoutes.home, label: 'Home' },
    { to: FrontSiteRoutes.documentsList, label: 'Search' },
  ];

  _renderElement(isLoading, error) {
    return (
      <>
        <Container fluid className="document-details-search-container">
          <Container>
            <SearchBar
              currentQueryString={this.state.searchQuery}
              onInputChange={this.onSearchInputChange}
              executeSearch={this.onSearchClick}
              placeholder={'Search for books...'}
              className="document-details-search-bar"
            />
          </Container>
        </Container>

        <Error error={error}>
          <Container className="document-details-container default-margin-top">
            <ILSParagraphPlaceholder isLoading={isLoading} lines={1}>
              <Breadcrumbs
                isLoading={isLoading}
                elements={this.breadcrumbs()}
                currentElement={
                  this.props.documentDetails.metadata
                    ? this.props.documentDetails.metadata.title
                    : null
                }
              />
            </ILSParagraphPlaceholder>
            <DocumentPanel />
          </Container>
          <Container className="document-tags">
            <ILSParagraphPlaceholder linesNumber={1} isLoading={isLoading}>
              <DocumentTags />
            </ILSParagraphPlaceholder>
          </Container>
          <Container className="section" fluid>
            <Container>
              <ILSParagraphPlaceholder linesNumber={20} isLoading={isLoading}>
                <DocumentMetadata />
              </ILSParagraphPlaceholder>
            </Container>
          </Container>
        </Error>
      </>
    );
  }

  render() {
    const { isLoading, error, data } = this.props;
    return this.renderElement(isLoading, error, data);
  }
}

DocumentsDetails.propTypes = {
  fetchDocumentsDetails: PropTypes.func.isRequired,
  renderElement: PropTypes.node,
};
