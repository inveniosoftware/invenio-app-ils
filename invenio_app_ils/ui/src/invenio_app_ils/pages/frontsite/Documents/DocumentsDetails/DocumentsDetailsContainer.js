import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Container } from 'semantic-ui-react';
import { DocumentMetadata } from './components';
import { goTo } from '../../../../history';
import { FrontSiteRoutes } from '../../../../routes/urls';
import { SearchBar } from '../../../../common/components/SearchBar';
import { DocumentPanel } from './components/DocumentPanel';
import { Error } from '../../../../common/components/Error';
import { Loader } from '../../../../common/components/Loader';
import { Breadcrumbs } from '../../components/Breadcrumbs';
import { DocumentTags } from './components/DocumentMetadata/components';

export class DocumentDetails extends Component {
  breadcrumbs = () => [
    { to: FrontSiteRoutes.home, label: 'Home' },
    { to: FrontSiteRoutes.documentsList, label: 'Search' },
  ];

  render() {
    return (
      <>
        <Container className="document-details-container default-margin-top">
          <Breadcrumbs
            elements={this.breadcrumbs()}
            currentElement={this.props.data.metadata.title}
          />
          <DocumentPanel />
        </Container>
        <Container className="document-tags">
          <DocumentTags tags={this.props.data.metadata.tags} />
        </Container>
        <Container className="section" fluid>
          <Container>
            <DocumentMetadata />
          </Container>
        </Container>
      </>
    );
  }
}

export default class DocumentsDetailsContainer extends Component {
  constructor(props) {
    super(props);
    this.fetchDocumentsDetails = this.props.fetchDocumentsDetails;
    this.state = { searchQuery: '' };
    this.renderElement = props.renderElement || this._renderElement;
  }

  componentDidMount() {
    this.unlisten = this.props.history.listen(location => {
      if (location.state && location.state.documentPid) {
        this.fetchDocumentsDetails(location.state.documentPid);
      }
    });
    this.fetchDocumentsDetails(this.props.match.params.documentPid);
  }

  componentWillUnmount() {
    this.unlisten();
  }

  onSearchClick = () => {
    const query = encodeURIComponent(this.state.searchQuery);
    goTo(FrontSiteRoutes.documentsListWithQuery(query));
  };

  onSearchInputChange = (value, event) => {
    this.setState({ searchQuery: event.target.value });
  };

  _renderElement(isLoading, error, data) {
    return (
      <>
        <Container fluid className="document-details-search-container">
          <Container>
            <SearchBar
              currentQueryString={this.state.searchQuery}
              onInputChange={this.onSearchInputChange}
              executeSearch={this.onSearchClick}
              placeholder={'Search for books...'}
              className="document-details-search-bar"
            />
          </Container>
        </Container>

        <Loader isLoading={isLoading}>
          <Error error={error}>
            <DocumentDetails data={data} />
          </Error>
        </Loader>
      </>
    );
  }

  render() {
    const { isLoading, error, data } = this.props;
    return this.renderElement(isLoading, error, data);
  }
}

DocumentsDetailsContainer.propTypes = {
  fetchDocumentsDetails: PropTypes.func.isRequired,
  renderElement: PropTypes.node,
};
