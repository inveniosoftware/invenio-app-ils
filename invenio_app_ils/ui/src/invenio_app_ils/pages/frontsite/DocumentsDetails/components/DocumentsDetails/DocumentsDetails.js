import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Loader, Error } from '../../../../../common/components';
import { DocumentMetadata } from '../';

export default class DocumentsDetails extends Component {
  render() {
    const { isLoading, error } = this.props;
    return (
      <Loader isLoading={isLoading}>
        <Error error={error}>
          <DocumentMetadata />
        </Error>
      </Loader>
    );
  }
}

DocumentsDetails.propTypes = {
  isLoading: PropTypes.bool.isRequired,
  data: PropTypes.object,
  hasError: PropTypes.bool.isRequired,
  error: PropTypes.object,
};
