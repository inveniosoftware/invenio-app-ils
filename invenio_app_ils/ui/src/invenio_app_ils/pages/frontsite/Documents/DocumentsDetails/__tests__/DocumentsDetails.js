import React from 'react';
import { shallow, mount } from 'enzyme';
import * as testData from '../../../../../../../../../tests/data/documents.json';

import DocumentsDetails from '../DocumentsDetails';
import { BrowserRouter } from 'react-router-dom';

jest.mock('../DocumentPanel', () => {
  return {
    DocumentPanel: () => null,
  };
});
jest.mock('../DocumentMetadata', () => {
  return {
    DocumentMetadata: () => null,
  };
});
jest.mock('../DocumentCirculation', () => {
  return {
    DocumentCirculation: () => null,
  };
});

jest.mock('../DocumentMetadata/components', () => {
  return {
    DocumentTags: () => null,
  };
});

describe('DocumentsDetailsContainer tests', () => {
  let component;
  afterEach(() => {
    if (component) {
      component.unmount();
    }
  });

  const routerHistory = {
    listen: () => () => {},
  };
  const routerUrlParams = {
    params: {
      documentPid: 111,
    },
  };

  it('should load the details component', () => {
    const component = shallow(
      <DocumentsDetails
        history={routerHistory}
        match={routerUrlParams}
        fetchDocumentsDetails={() => {}}
        documentDetails={{ metadata: testData[0] }}
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('should fetch item details on mount', () => {
    const mockedFetchDocumentsDetails = jest.fn();
    component = mount(
      <BrowserRouter>
        <DocumentsDetails
          history={routerHistory}
          match={routerUrlParams}
          fetchDocumentsDetails={mockedFetchDocumentsDetails}
          documentDetails={{ metadata: testData[0] }}
        />
      </BrowserRouter>
    );
    expect(mockedFetchDocumentsDetails).toHaveBeenCalledWith(
      routerUrlParams.params.documentPid
    );
  });
});
