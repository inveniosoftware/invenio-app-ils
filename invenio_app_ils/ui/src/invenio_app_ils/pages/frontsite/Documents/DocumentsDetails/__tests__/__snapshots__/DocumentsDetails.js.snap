// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocumentsDetailsContainer tests should load the details component 1`] = `
<Fragment>
  <Container
    className="document-details-search-container"
    fluid={true}
  >
    <Container>
      <SearchBar
        className="document-details-search-bar"
        currentQueryString=""
        executeSearch={[Function]}
        onInputChange={[Function]}
        placeholder="Search for books..."
      />
    </Container>
  </Container>
  <Error>
    <Container
      className="document-details-container default-margin-top"
    >
      <ILSParagraphPlaceholder
        lines={1}
      >
        <Breadcrumbs
          currentElement="The Gulf: The Making of An American Sea"
          elements={
            Array [
              Object {
                "label": "Home",
                "to": "/",
              },
              Object {
                "label": "Search",
                "to": "/search",
              },
            ]
          }
        />
      </ILSParagraphPlaceholder>
      <DocumentPanel />
    </Container>
    <Container
      className="document-tags"
    >
      <ILSParagraphPlaceholder
        linesNumber={1}
      >
        <DocumentTags />
      </ILSParagraphPlaceholder>
    </Container>
    <Container
      className="section"
      fluid={true}
    >
      <Container>
        <ILSParagraphPlaceholder
          linesNumber={20}
        >
          <DocumentMetadata />
        </ILSParagraphPlaceholder>
      </Container>
    </Container>
  </Error>
</Fragment>
`;
